activemodel/lib/active_model/validations/numericality.rb

```diff
@@ -92,21 +92,17 @@ def parse_as_number(raw_value)
           raw_value
         elsif is_integer?(raw_value)
           raw_value.to_i
-        elsif is_decimal?(raw_value) && !is_hexadecimal_literal?(raw_value)
-          BigDecimal(raw_value)
+        elsif !is_hexadecimal_literal?(raw_value)
+          BigDecimal(Kernel.Float(raw_value), Float::DIG)
         end
       end
 
       def is_integer?(raw_value)
         INTEGER_REGEX.match?(raw_value.to_s)
       end
 
-      def is_decimal?(raw_value)
-        DECIMAL_REGEX.match?(raw_value.to_s)
```

Comment:
We can remove `DECIMAL_REGEX` too.
https://github.com/rails/rails/pull/35010/files#diff-58f66170fe2bce77e8a8b3c69b77a0d6L13